pyqtgraph/palette.py

```diff
@@ -0,0 +1,414 @@
+    ===============  ====================================================================
+     **Arguments:**
+    identifier       'system' (default): Colors are based on current Qt QPalette
+                     'legacy': The color scheme of previous versions of PyQtGraph
+                     'monochrome' ['color identifier']: Creates a palette that imitates
+                        a monochrome computer monitor.
+                        'color identifier' can be one of 
+                        'green', 'amber', 'blue', 'red', 'lavender', 'pink'
+                        or a tuple of relative R,G,B contributions in range 0.0 to 1.0
+
+                     {dictionary}: full palette specification, see palette.py for details
+    ===============  ====================================================================
+    """    
+    if identifier == 'system':
+        pal = Palette()
+        return pal # default QPalette based settings
+    if identifier == 'monochrome':
+        pal = Palette()
+        pal.setMonochrome( *args )
+        return pal
+    if identifier in PALETTE_DEFINITIONS:
+        info = PALETTE_DEFINITIONS[identifier].copy()
+        sampling_info = info.pop('colormap_sampling', None)
+        pal = Palette( cmap=sampling_info, colors=info )
+        return pal
+    raise KeyError("Unknown palette identifier '"+str(identifier)+"'")
+
+class Palette(object):
+    """
+    A Palette object provides a set of colors that can conveniently applied 
+    to the PyQtGraph color scheme.
+    It specifies at least the following colors, but additional one can be added:
+    Primary colors:
+      'b', 'c', 'g', 'y', 'r', 'm'
+    Gray scale:
+      'k', 'd', 'l', 'w'  ranging from black to white
+      's' slate gray
+    System colors:
+      'gr_bg', 'gr_fg', 'gr_txt'  graph background, foreground and text colors
+      'gr_wdw'  window background color
+      'gr_reg'  partially transparent region shading color
+      'gr_acc'  accent for UI elements
+      'gr_hlt'  highlight for selected elements
+    Plot colors:
+      'p0' to 'p7'  typically sampled from a ColorMap
+    """
+    def __init__(self, cmap=None, colors=None ):
+        super().__init__()
+        self.palette = fn.COLOR_REGISTRY.defaultColors()
+        # self.palette  = { # populate dictionary of QColors with legacy defaults
+        #     'b': QtGui.QColor(  0,  0,255,255), 'g': QtGui.QColor(  0,255,  0,255), 
+        #     'r': QtGui.QColor(255,  0,  0,255), 'c': QtGui.QColor(  0,255,255,255),
+        #     'm': QtGui.QColor(255,  0,255,255), 'y': QtGui.QColor(255,255,  0,255),
+        #     'k': QtGui.QColor(  0,  0,  0,255), 'w': QtGui.QColor(255,255,255,255),
+        #     'd': QtGui.QColor(150,150,150,255), 'l': QtGui.QColor(200,200,200,255), 
+        #     's': QtGui.QColor(100,100,150,255)
+        # }
+        # self.addDefaultRamp()
+        self.dark = None # is initially set when assigning system palette
```

Comment:
instead of setting to None, we can set default to what the current setup is:

https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/Qt.py#L417-L420

```python
app = QtWidgets.QApplication.instance()
self.dark = app.property('darkMode')
```

This way you shouldn't have to worry about the case of `None`

Reply:
For the moment, each palette has a `dark` property, which currently simply represents that the background color is darker than (127,127,127). Since we are looking only at the plot palette here, this is not strongly linked to the system palette. PyQtGraph has been plotting dark plots in a light window for a long time.

As the comment says, this gets initialized immediately, so there is nothing to worry about. But not setting it makes the linter complain...

So I think this is okay here. We might want some method somewhere that sets the initial plot palette based on the system mode... But that would happen somewhere else. And for continuity, I think PyQtGraph should always start up with a dark plotting palette until instructed otherwise.